code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstanceCreate.java

```diff
@@ -88,6 +89,34 @@ protected boolean shouldStart(Instance instance) {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private List<Volume> processManagedVolumes(ProcessState state, Instance instance) {
+        List<Volume> volumes = new ArrayList<Volume>();
+        Map<String, Object> dataVolumeMounts = DataAccessor.field(instance, InstanceConstants.FIELD_DATA_VOLUME_MOUNTS, Map.class);
+        if (dataVolumeMounts != null) {
+            List<String> dataVolumes = DataAccessor.field(instance, InstanceConstants.FIELD_DATA_VOLUMES, List.class);
```

Comment:
Can you early exist, so just return volumes and not have the whole block indented.